ophyd/controls/areadetector/detectors.py

```diff
@@ -39,9 +41,138 @@
            'URLDetector',
            ]
 
+def set_and_wait(signal, val):
+    """
+    Set a signal to a value and wait until it reads correctly.
+
+    There are cases where this would not work well, so it should be revisited.
+    """
+    signal.put(val)
+    while signal.get() != val:
+        ttime.sleep(0.1)
+        logger.info("Waiting for %s to be set...", signal.name)
+
 
 class DetectorBase(ADBase):
-    pass
+    "This base class handles the staging, unstaging, and triggering."
+
+    # OphydObj subscriptions
+    _SUB_ACQ_DONE = 'acq_done'
+    _SUB_TRIGGER_DONE = 'trigger_done'
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # TODO Should we make these settings customizable in the init?
+        # If so, what API?
+
+        # settings
+        self._stage_sigs = {self.cam.acquire: 0,  # If acquiring, stop.
+                            self.cam.image_mode: 1  # 'Multiple' mode
+                           }
+
+        self._staged = False
+
+        self._acquisition_signal = self.cam.acquire  # for generality's sake
+        self._acquisition_signal.subscribe(self._acquire_changed)
+
+    def stage(self):
```

Comment:
Except for the last line, this code (and all the code for `unstage`) is completely generic. Maybe it should move into `OphydDevice`.